### PR TITLE
Remove bounds on cyclo-stationary AR(1)

### LINKED ISSUE #472: Bounds to AR1 process in MESMER-M
In MESMER-M we estimate the parameters of the AR1 process to generate local temporally correlated variability. In principle, this is similar as in MESMER, except that we want AR parameters for each month individually. This results from the assumption that June might depend on May differently than July does on June. As a consequence we need to implement an in-house fitting method. Which looks like this:

```
def _fit_autoregression_monthly_np(data_month, data_prev_month):
    
    def lin_func(x, a, b):
        return a * x + b
    
    slope, intercept = curve_fit(lin_func, 
                     data_prev_month, # independent variable
                     data_month, # dependent variable
                     bounds=([-1,-np.inf], [1, np.inf]))[0]
    
    return slope, intercept
```

We fit a linear function using each value of the previous month as predictor for the next month. Herein, Shruti set the bounds for the slope to [-1, 1]. This is needed because when we later adjust the covariance we do $\Sigma_{adjusted} = \sqrt{1- \gamma_i^2} \cdot \sqrt{1- \gamma_j^2} \cdot \Sigma_{empirical}$ with $\gamma$ being the AR slope at each gridpoint and thus $\gamma \in [-1, 1]$. And indeed, without this bound, some slopes lie outside this interval.

Now, I don't see this restriction in MESMER. Also, in our example, it does not happen that the slope goes outside of this interval. Now my question is, should we worry about these slopes in MESMER-M? I still don't completely understand the rationale behind the adjustment so I'm sceptical about enforcing these bounds uninformed.

### PR DESCRIPTION
**Pull request by veni-vidi-vici-dormivi**
- [x] Closes #472 
 - [x] Tests added
 - [x] Fully documented, including `CHANGELOG.rst`

We want to return the residuals of the cycle-stationary AR(1) process to fit the covariance matrix on these. This removes the need for bounds in the fit as well as `predict_auto_regression`.

### CODE REVIEW

**Review by mathause — COMMENTED** [review]
Thanks! Now we no longer need to call `_adjust_ecov_ar1_np` - right? Will you remove that in another PR? Or is that removed implicitly?

**Comment by veni-vidi-vici-dormivi** [comment]
> Now we no longer need to call _adjust_ecov_ar1_np - right? Will you remove that in another PR? Or is that removed implicitly?

We only called the adjustment in the notebook, not in any of the functions, so I removed it in the notebook and the example I am preparing but no changes are needed in the mesmer code.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -63,6 +63,11 @@ that this led to some numerical changes compared to the MESMER-M publication
 - Implement functions performing the monthly (cyclo-stationary) auto-regression and adapt these functions to
   work with xarray. This includes extracting the drawing of spatially correlated innovations to a
   stand-alone function. (`#473 <https://github.com/MESMER-group/mesmer/pull/473>`_)
+- Remove the bounds of -1 and 1 on the slope of the cyclo-stationary AR(1) process. This bound is not necessary
+  since cyclo-stationarity is also given if the slopes of a few months are (slightly) larger than one. This however
+  leads to the fact that the adjustment of the covariance matrix with these slopes is not possible anymore. We
```
[the remark this answers is not in the document]
Comment (veni-vidi-vici-dormivi) [reply CHANGELOG.rst:68]: Good point

File: tests/unit/test_auto_regression.py [hunk 2]
```diff
@@ -597,33 +597,46 @@ def test_fit_auto_regression_np(lags):
...
+    np.testing.assert_allclose(intercept, intercept_fit)
+    np.testing.assert_allclose(residuals, expected_resids, atol=1e-6)
 
 
-@pytest.mark.parametrize("slope", [2.0, -2])
-def test_fit_autoregression_monthly_np_outside_bounds(slope):
-    # test that given a slope outside the bounds of -1 and 1, the slope is clipped
+@pytest.mark.parametrize("intercept", [1.0, -4.0])
+@pytest.mark.parametrize("slope", [0.2, -0.3])
+@pytest.mark.parametrize("variance", [1, 0.5])
```
[the remark this answers is not in the document]
Comment (veni-vidi-vici-dormivi) [reply tests/unit/test_auto_regression.py:621]: Ah true thanks.

File: tests/unit/test_auto_regression.py [hunk 3]
```diff
@@ -597,33 +597,46 @@ def test_fit_auto_regression_np(lags):
...
     result = mesmer.stats._auto_regression._fit_auto_regression_monthly_np(
         cur_month, prev_month
     )
-    expected_slope = 1.0 * np.sign(slope)
-    np.testing.assert_allclose(result[0], expected_slope)
+    slope_fit, intercept_fit, residuals = result
+
+    np.testing.assert_allclose(slope, slope_fit, atol=1e-1)
+    np.testing.assert_allclose(intercept, intercept_fit, atol=1e-1)
+    np.testing.assert_allclose(np.var(residuals), variance, rtol=1)
```
[the remark this answers is not in the document]
Comment (veni-vidi-vici-dormivi) [reply tests/unit/test_auto_regression.py]: Of course, thanks a lot.

[the remark this answers is not in the document]
Comment (veni-vidi-vici-dormivi) [reply tests/unit/test_auto_regression.py]: And yes it does get us closer. Now also at `atol=1e-1`.

File: tests/unit/test_auto_regression.py [hunk 4]
```diff
@@ -597,33 +597,44 @@ def test_fit_auto_regression_np(lags):
 
 @pytest.mark.parametrize("intercept", [1.0, -4.0])
 @pytest.mark.parametrize("slope", [0.2, -0.3])
-def test_fit_autoregression_monthly_np(slope, intercept):
+def test_fit_autoregression_monthly_np_no_noise(slope, intercept):
```
Comment (mathause) [inline tests/unit/test_auto_regression.py:600]: Good idea - that's a descriptive name.

**Review by mathause — APPROVED** [review]
Thanks! Good statistical detective work :slightly_smiling_face:

**Pull request merged**